
# HG changeset patch
# User serge-sans-paille <sguelton@mozilla.com>
# Date 1699342987 0
# Node ID 201fc291a2c0f9c8fd86c7bf5647cf7d179ca563
# Parent  b2826966ef8a149fffb1e8ca7ea0bd235975e19b
Bug 1858062 - Remove reference to distutils in mozboot r=saschanaz

distutils is no longer available in Python 3.12, which means we cannot
require it on bootstrap, otherwise we cannot bootstrap from Py 3.12.

We already require setuptools as part of mach's requirements, so it's
fine to depend on setuptools elsewhere.

Differential Revision: https://phabricator.services.mozilla.com/D190537

diff --git a/python/mozboot/mozboot/base.py b/python/mozboot/mozboot/base.py
--- a/python/mozboot/mozboot/base.py
+++ b/python/mozboot/mozboot/base.py
@@ -165,23 +165,16 @@ class BaseBootstrapper(object):
 
     def validate_environment(self):
         """
         Called once the current firefox checkout has been detected.
         Platform-specific implementations should check the environment and offer advice/warnings
         to the user, if necessary.
         """
 
-    def suggest_install_distutils(self):
-        """Called if distutils.{sysconfig,spawn} can't be imported."""
-        print(
-            "Does your distro require installing another package for distutils?",
-            file=sys.stderr,
-        )
-
     def suggest_install_pip3(self):
         """Called if pip3 can't be found."""
         print(
             "Try installing pip3 with your system's package manager.", file=sys.stderr
         )
 
     def install_system_packages(self):
         """
diff --git a/python/mozboot/mozboot/bootstrap.py b/python/mozboot/mozboot/bootstrap.py
--- a/python/mozboot/mozboot/bootstrap.py
+++ b/python/mozboot/mozboot/bootstrap.py
@@ -487,32 +487,16 @@ class Bootstrapper(object):
                 # The mozconfig file is created by user.
                 self._show_mozconfig_suggestion(raw_mozconfig)
         elif raw_mozconfig:
             # No mozconfig file exists yet
             self._write_default_mozconfig(raw_mozconfig)
 
     def _validate_python_environment(self, topsrcdir):
         valid = True
-        try:
-            # distutils is singled out here because some distros (namely Ubuntu)
-            # include it in a separate package outside of the main Python
-            # installation.
-            import distutils.spawn
-            import distutils.sysconfig
-
-            assert distutils.sysconfig is not None and distutils.spawn is not None
-        except ImportError as e:
-            print("ERROR: Could not import package %s" % e.name, file=sys.stderr)
-            self.instance.suggest_install_distutils()
-            valid = False
-        except AssertionError:
-            print("ERROR: distutils is not behaving as expected.", file=sys.stderr)
-            self.instance.suggest_install_distutils()
-            valid = False
         pip3 = to_optional_path(which("pip3"))
         if not pip3:
             print("ERROR: Could not find pip3.", file=sys.stderr)
             self.instance.suggest_install_pip3()
             valid = False
         if not valid:
             print(
                 "ERROR: Your Python installation will not be able to run "
diff --git a/python/mozboot/mozboot/debian.py b/python/mozboot/mozboot/debian.py
--- a/python/mozboot/mozboot/debian.py
+++ b/python/mozboot/mozboot/debian.py
@@ -12,23 +12,16 @@ class DebianBootstrapper(LinuxBootstrapp
     def __init__(self, distro, version, dist_id, codename, **kwargs):
         BaseBootstrapper.__init__(self, **kwargs)
 
         self.distro = distro
         self.version = version
         self.dist_id = dist_id
         self.codename = codename
 
-    def suggest_install_distutils(self):
-        print(
-            "HINT: Try installing distutils with "
-            "`apt-get install python3-distutils`.",
-            file=sys.stderr,
-        )
-
     def suggest_install_pip3(self):
         print(
             "HINT: Try installing pip3 with `apt-get install python3-pip`.",
             file=sys.stderr,
         )
 
     def install_packages(self, packages):
         try:
diff --git a/python/mozboot/setup.py b/python/mozboot/setup.py
--- a/python/mozboot/setup.py
+++ b/python/mozboot/setup.py
@@ -1,13 +1,16 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from distutils.core import setup
+try:
+    from setuptools import setup
+except ImportError:
+    from distutils.core import setup
 
 VERSION = "0.1"
 
 setup(
     name="mozboot",
     description="System bootstrap for building Mozilla projects.",
     license="MPL 2.0",
     packages=["mozboot"],

